hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/HoodieTimelineArchiveLog.java

```diff
@@ -198,14 +200,20 @@ public boolean archiveIfRequired(HoodieEngineContext context) throws IOException
     // If metadata table is enabled, do not archive instants which are more recent that the latest synced
     // instant on the metadata table. This is required for metadata table sync.
     if (config.useFileListingMetadata()) {
-      Option<String> lastSyncedInstantTime = table.metadata().getSyncedInstantTime();
-      if (lastSyncedInstantTime.isPresent()) {
-        LOG.info("Limiting archiving of instants to last synced instant on metadata table at " + lastSyncedInstantTime.get());
-        instants = instants.filter(i -> HoodieTimeline.compareTimestamps(i.getTimestamp(), HoodieTimeline.LESSER_THAN,
-            lastSyncedInstantTime.get()));
-      } else {
-        LOG.info("Not archiving as there is no instants yet on the metadata table");
-        instants = Stream.empty();
+      try (HoodieTableMetadata tableMetadata = HoodieTableMetadata.create(table.getContext(), config.getBasePath(), "/tmp/",
```

Comment:
do we just pass `/tmp` always? . `HoodieTableMetadata#create()` should just take `HoodieMetadataConfig` as arg

Reply:
This is the spillableMapPath. Can be change to config.getSpillableMapBasePath().